### ISSUE TITLE
Crash when testing Libtorch example

### ISSUE THREAD

**Issue by Laurick1** [pytorch/pytorch#129819]
### 🐛 Describe the bug

Hello,

I tried in my software the example giving by Libtorch (https://pytorch.org/tutorials/advanced/cpp_autograd.html) and it crashed. I can't figure out why it is happening

`#include <torch/torch.h>

auto model = torch::nn::Linear(4, 3);

auto input = torch::randn({3, 4}).requires_grad_(true);
auto output = model(input);

// Calculate loss
auto target = torch::randn({3, 3});
auto loss = torch::nn::MSELoss()(output, target);

// Use norm of gradients as penalty
auto grad_output = torch::ones_like(output);
auto gradient = torch::autograd::grad({output}, {input}, /*grad_outputs=*/{grad_output}, /*create_graph=*/true)[0];
auto gradient_penalty = torch::pow((gradient.norm(2, /*dim=*/1) - 1), 2).mean();

// Add gradient penalty to loss
auto combined_loss = loss + gradient_penalty;
combined_loss.backward();

std::cout << input.grad() << std::endl;`

I tried to apply the correct device on the model and the tensors but it didn't change anything ; I tried to add requires_grad_(true) but it didn't change anything either.

Have you ever faced this problem? And how did you manage to solve it?

Thank you in advance,

Laurick

### Versions

C++17
Visual Studio 2019
Libtorch : 2.0.0+cu118

**Comment by Laurick1** [comment]
Before crashing, the software returns the following values:

Input:
 0.6896  0.0109 -0.8927 -0.6019
-1.2692 -0.3748  1.2181  1.2204
 1.5659  0.6650  0.2665  0.3639
[ CUDAFloatType{3,4} ]

Target:
 1.1027  1.2941  0.5516
-1.4952 -0.2794  0.0011
-0.2567  1.1541  0.6320
[ CUDAFloatType{3,3} ]

Output:
 0.1708  0.4900 -0.3538
-0.1719  0.5131 -0.8069
 0.8201 -0.4959 -0.9887
[ CUDAFloatType{3,3} ]

grad_output:
 1  1  1
 1  1  1
 1  1  1
[ CUDAFloatType{3,3} ]

**Comment by malfet** [comment]
Hmm, 2.0.0 were release a while back, and lots of folks were able to compile libtorch example since then, which suggests something specific with your local setup. Do you mind posting the same question to https://discuss.pytorch.org/
Would it be possible to collect/post a crash backtrace here?

**Comment by taras-janea** [comment]
The provided code snippet executes successfully on the latest PyTorch - no crash.
@Laurick1, if the issue is still reproducible on your setup with the latest PyTorch version, then please share:
- complete stack trace of the crash;
- output of the [script](https://raw.githubusercontent.com/pytorch/pytorch/main/torch/utils/collect_env.py).

**Comment by taras-janea** [comment]
Closing this issue due to inactivity. If you continue to experience this problem or have any related concerns, please don't hesitate to open a new issue.
